CMakeLists.txt

```
@@ -57,6 +57,12 @@ include(cmake/NVBenchInstallRules.cmake)
 include(cmake/NVBenchUtilities.cmake)
 include(cmake/NVBenchLibhipcxx.cmake)
 
+# synchronize HIP and CUDA architectures
+if (DEFINED CMAKE_HIP_ARCHITECTURES AND NOT CMAKE_CUDA_ARCHITECTURES)
```

Could we add this sentence to detect rocm arch automatically ?

```suggestion
    EXECUTE_PROCESS(COMMAND bash -c "/opt/rocm/bin/rocminfo | grep -o -m1 'gfx.*'" 
    OUTPUT_VARIABLE CMAKE_HIP_ARCHITECTURES
    OUTPUT_STRIP_TRAILING_WHITESPACE)
      
if (DEFINED CMAKE_HIP_ARCHITECTURES AND NOT CMAKE_CUDA_ARCHITECTURES)
```

---

I would prefer not to have that here in hipBench. This can be placed inside of a user script.

There is a couple of further issues with this:

* ROCm is not always installed into `/opt/rocm`.
* Build system and test/bench system may not be the same.